app/assets/javascripts/views/response_list_view.js

```diff
@@ -22,7 +26,42 @@ ELMO.Views.ResponseListView = class ResponseListView extends ELMO.Views.Applicat
 
   showExportCsvModal(event) {
     event.preventDefault();
-    this.$('#export-csv-modal').modal('show');
+
+    // Reset defaults in case the modal is shown several times.
+    $('#export-count-warning').hide();
+    $('#export-count-error').hide();
+    $("input[type=submit]").prop("disabled", false);
+
+    // Calculate how many responses will actually be exported.
+    let responsesForm = $('.index-table-wrapper form');
+    let checked = responsesForm.find('input.batch_op:checked');
+    let selectAll = responsesForm.find('input[name=select_all_pages]');
+    let shouldExportAll = checked.length === 0 || selectAll.val();
+    let exportCount = shouldExportAll ? $('.index-table-wrapper').data('entries') : checked.length;
+    $('#export-options-summary').text(I18n.t('response.export_options.summary', { count: exportCount }));
+
```

Comment:
For my learnings... why is all of this here instead of the initialize function in the modal?

Reply:
Partly for messy reasons (stuff was there originally and I added to it), partly for convenience (I need access to the form OUTSIDE of the modal, so by default I think `$` is scoped to the modal itself and I'd need to do something more complex to search for the outer form.

I realized at the end of this work that it could be moved but the DOM is so fragile I didn't want to test everything over again 🤷

Reply:
Well I def support that!